web/src/hooks/redux.js

```diff
@@ -30,11 +30,13 @@ export function useCurrentZoneHistoryDatetimes() {
 // the graph (when not inferable from historyData timestamps).
 export function useCurrentZoneHistoryEndTime() {
   const customDatetime = useCustomDatetime();
-  const gridDatetime = useSelector(state => (state.data.grid || {}).datetime);
+  const gridDatetime =
+    useSelector((state) => (state.data.grid || {}).datetime) ||
+    moment().subtract(2, 'hour').startOf('hour');
```

Comment:
Would be nice if we could get the latest timestamp from data if possible :)

Reply:
The issue is that if the state becomes outdated, I would think that we want it to show a gap in the graph
(see this comment: https://github.com/tmrowco/electricitymap-contrib/pull/3216/files/4a5a80045c8ea4492c228b46747a750808e09ec9#diff-5754e0648333a5b23fb97471161519bbe522f06ed95da9668d10c2a734559642R28-R30)

What do you think is the best solution here?

Reply:
Ah, I guess that makes sense - let's stick with this solution then :)